src/main/scala/org/apache/spark/sql/pulsar/CachedPulsarClient.scala

```diff
@@ -62,7 +65,11 @@ private[pulsar] object CachedPulsarClient extends Logging {
       .removalListener(removalListener)
       .build[Seq[(String, Object)], Client](cacheLoader)
 
-  private def createPulsarClient(pulsarConf: ju.Map[String, Object]): Client = {
+  def createPulsarClient(
```

Comment:
This method is exposed, why? I guess that you would like to use it in `PulsarMetadataReader.scala`.

Reply:
This had been exposed so that the test of the class could access it. My original idea was not to fire up an authentication-enabled Pulsar cluster for this purpose (just for unit testing this class), I wanted check the authentication parameters are set properly in the client builder only.